src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,200 @@
+
+public class IntakeIOTalonFX implements IntakeIO {
+  private TalonFX rightRollerMotor;
+  private TalonFX leftRollerMotor;
+  private TalonFX drumMotor;
+  private final DigitalInput rightRollerIRSensor;
+  private final DigitalInput leftRollerIRSensor;
+  private final DigitalInput drumIRSensor;
+
+  private double rightRollerRequestedVelocity;
+  private double leftRollerRequestedVelocity;
+  private double drumRequestedVelocity;
+
+  private VelocityTorqueCurrentFOC rightRollerVelocityRequest;
+  private VelocityTorqueCurrentFOC leftRollerVelocityRequest;
+  private VelocityTorqueCurrentFOC drumVelocityRequest;
+
+  private StatusSignal<Double> rightRollerVelocityStatusSignal;
+  private StatusSignal<Double> leftRollerVelocityStatusSignal;
+  private StatusSignal<Double> drumVelocityStatusSignal;
+  // status signals for stator current
+  private StatusSignal<Double> rightRollerStatorCurrentStatusSignal;
+  private StatusSignal<Double> leftRollerStatorCurrentStatusSignal;
+  private StatusSignal<Double> drumStatorCurrentStatusSignal;
+
+  private StatusSignal<Double> rightRollerSupplyCurrentStatusSignal;
+  private StatusSignal<Double> leftRollerSupplyCurrentStatusSignal;
+  private StatusSignal<Double> drumSupplyCurrentStatusSignal;
+
+  // tunable numbers for roller and drum pid
+  private final TunableNumber rollerMotorsKP =
+      new TunableNumber("Intake/rollerMotorsKP", IntakeConstants.INTAKE_ROLLER_MOTORS_KP);
+
+  private final TunableNumber rollerMotorsKI =
+      new TunableNumber("Intake/rollerMotorsKI", IntakeConstants.INTAKE_ROLLER_MOTORS_KI);
+
+  private final TunableNumber rollerMotorsKD =
+      new TunableNumber("Intake/rollerMotorsKD", IntakeConstants.INTAKE_ROLLER_MOTORS_KD);
+
+  private final TunableNumber rollerMotorsKS =
+      new TunableNumber("Intake/rollerMotorsKS", IntakeConstants.INTAKE_ROLLER_MOTORS_KS);
+
+  private final TunableNumber drumMotorKP =
+      new TunableNumber("Intake/drumMotorKP", IntakeConstants.INTAKE_DRUM_MOTOR_KP);
+  private final TunableNumber drumMotorKI =
+      new TunableNumber("Intake/drumMotorKI", IntakeConstants.INTAKE_DRUM_MOTOR_KI);
+  private final TunableNumber drumMotorKD =
+      new TunableNumber("Intake/drumMotorKD", IntakeConstants.INTAKE_DRUM_MOTOR_KD);
+
+  private final TunableNumber drumMotorKS =
+      new TunableNumber("Intake/drumMotorKS", IntakeConstants.INTAKE_DRUM_MOTOR_KS);
+
+  public IntakeIOTalonFX() {
+    rightRollerIRSensor = new DigitalInput(IntakeConstants.INTAKE_RIGHT_ROLLER_IR_SENSOR_ID);
+    leftRollerIRSensor = new DigitalInput(IntakeConstants.INTAKE_LEFT_ROLLER_IR_SENSOR_ID);
+    drumIRSensor = new DigitalInput(IntakeConstants.INTAKE_DRUM_IR_SENSOR_ID);
+
+    rightRollerVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    leftRollerVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    drumVelocityRequest = new VelocityTorqueCurrentFOC(0);
+
+    rightRollerVelocityStatusSignal = rightRollerMotor.getRotorVelocity();
```

Comment:
I think we should invoke getVelocity which accounts for the sensor to mechanism ratio and, therefore, returns the velocity of the roller, rather than the velocity of the motor. To configure the sensor to mechanism ratio in the config method:

```
rollerConfig.Feedback.SensorToMechanismRatio = xxx;
```

Reply:
This is useful because we don't actually care about the velocity of the Falcon shaft. We do care about the velocity of the roller. So, if we end up changing gear ratios, we only have to change this constant and all the tuning for a specific velocity is still valid.